crypto/provider_conf.c

```diff
@@ -313,3 +313,49 @@ void ossl_provider_add_conf_module(void)
     OSSL_TRACE(CONF, "Adding config module 'providers'\n");
     CONF_module_add("providers", provider_conf_init, NULL);
 }
+
+static CONF *ossl_conf_from_params(OSSL_LIB_CTX *ctx, OSSL_PARAM *params)
+{
+    CONF *my_conf = NCONF_new_ex(ctx, NCONF_default());
+    BIO *conf_bio = BIO_new(BIO_s_mem());
+    int ok = 0;
+    int i;
+
+    if (BIO_printf(conf_bio, "[%s]\nactivate = 1\n", "my_sect") < 0)
+        goto err;
+
+    for (i = 0; params[i].key != NULL; i++) {
+        if (params[i].data_type != OSSL_PARAM_UTF8_STRING)
+            continue;
+        if (BIO_printf(conf_bio, "%s = %s\n", params[i].key, (char *)params[i].data) <= 0)
+            goto err;
+    }
+
+    if (NCONF_load_bio(my_conf, conf_bio, NULL) != 1)
+        goto err;
+
+    ok = 1;
+err:
+    if (ok != 1) {
+        NCONF_free(my_conf);
+        my_conf = NULL;
+    }
+    BIO_free(conf_bio);
+
+    return my_conf;
+}
+
+int OSSL_PROVIDER_activate(OSSL_LIB_CTX *libctx, const char *name,
+                                OSSL_PARAM *params)
+{
+    CONF *my_conf = ossl_conf_from_params(libctx, params);
+    int ret = 0;
+
+    if (my_conf == NULL)
+        return 0;
+
+    ret = provider_conf_load(libctx, name, "my_sect", my_conf);
+    NCONF_free(my_conf);
+
+    return ret;
+}
```

Comment:
Without having gone into details too much yet:  the  name `OSSL_PROVIDER_activate()` seems to be a very misleading choice given the fact that

a) it is intended to be some alternative to `OSSL_provider_load()`
b) there already exists an unrelated function  `ossl_provider_activate()` where the name seems to be more fitting.

https://github.com/openssl/openssl/blob/e3d897d3fa3b48bb835fab0665a435469beea7ae/crypto/provider_core.c#L1274-L1292

How about one of the following names?
* `OPENSSL_provider_load_with_params()`
* `OPENSSL_provider_load_from_params()`

Reply:
I thought about this or similar naming but current OSSL_PROVIDER_load returns a provider handle and doesn't give me any opportunity to add config options (at least I don't see them). But I don't insist on the name.

The implementation I provided is intended to load the provider and activate it.

Reply:
Resolving this thread because I have implemented a normal way to deal with it